bundle/config/bundle.go

```diff
@@ -43,4 +43,8 @@ type Bundle struct {
 
 	// Overrides the compute used for jobs and other supported assets.
 	ComputeID string `json:"compute_id,omitempty"`
+
+	// FailIfRunning specifies whether to fail the deployment if there are
+	// running jobs or pipelines in the workspace. Defaults to false.
+	FailIfRunning bool `json:"fail_if_running,omitempty"`
```

Comment:
The place of this field in the overall config seems odd:
```yaml
bundle:
  fail_if_running: true
```

It's not clear what this pertains to. What would be a location to make its use more clear?

Reply:
@pietern we can have a `Deployment` similar to `Lock` one we have and put it under there?

Reply:
Sounds good -- we can consider moving the lock related fields there as well.

W.r.t. the name itself, it is ambiguous what is running: jobs or the deployment itself.

Alternative: `fail_on_active_runs` or `fail_if_runs_active` or something.